### PR TITLE
Use helper functions from ignition common and math

### PR DESCRIPTION
**Pull request by nkoenig**
Signed-off-by: Nate Koenig <nate@openrobotics.org>

# 🎉 New feature

This deprecates duplicate functions found in libsdformat in favor of the functions found in ignition common.

## Test it

The tests should all pass.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [x] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -111,7 +111,7 @@ if (BUILD_SDF)
 
   ########################################
   # Find ignition common
-  ign_find_package(ignition-common5 COMPONENTS graphics REQUIRED_BY usd)
+  ign_find_package(ignition-common5 COMPONENTS graphics REQUIRED)
```
Comment (azeey) [inline CMakeLists.txt:114]: Is there a way to require only the core of ign-common and not the graphics component? I think the graphics component brings too many dependencies that we wouldn't want to impose on SDFormat users.

Even the core of ign-common brings a dependency on `libdl`, which I think is an unnecessary constraint for SDFormat. As @chapulina suggested in https://github.com/ignitionrobotics/sdformat/pull/911#discussion_r838737967, perhaps we should move some of the functions used in this PR to `ign-utils`.

**Review by chapulina — CHANGES_REQUESTED** [review]
It's not worth it bringing `ign-common` as a dependency just to use a few short helper functions. There's some discussion in https://github.com/ignitionrobotics/sdformat/issues/351 about why bringing that dependency is problematic.

**Comment by nkoenig** [comment]
> It's not worth it bringing `ign-common` as a dependency just to use a few short helper functions. There's some discussion in #351 about why bringing that dependency is problematic.

I've opened two other pull requests that replaces Console and Filesystem with the equivalents in ign-common.

File: CMakeLists.txt (hunk 1 above)
Comment (nkoenig) [reply CMakeLists.txt]: I've added a find package for the just the main part of common. There are a couple other pull requests that also make use of ign-common's Filesystem and Console.

**Pull request closed without merging**